tap_chromedata/streams.py

```diff
@@ -7,518 +8,327 @@
-                        with ZipFile(flo) as archive:
-                            with archive.open('DeepLink.txt') as fd:
-                                data=fd.readlines()
-                                for j in range(len(data)):
-
-                                    data[j]=data[j].decode('utf-8')
-                                    tildcount=0
-                                    if '\r\n' in data[j]:
-                                        data[j]=data[j].replace('\r\n','')
-                                    for i in range(len(data[j])):
-                                        if data[j][i]=='~':
-                                            if tildcount==0:
-                                                tildcount+=1
-                                            else:
-                                                tildcount-=1
-                                        if data[j][i]==",":
-                                            if tildcount==1:
-                                                data[j]=data[j][:i]+' '+data[j][i+1:]
-                                    data[j] = data[j].split(',')
-                                    
-                                colnames=data[0]
-                                for i in range(len(colnames)):
-                                    if '~' in colnames[i]:
-                                        colnames[i]=colnames[i].replace('~','')
-                                    colnames[i]=camel_to_snake(colnames[i])
-                                for j in range(1,len(data)):
-                                    row_dict={}
-                                    for i in range(0,len(data[j])):
-                                        if data[j][i].isnumeric():
-                                            row_dict[colnames[i]]=int(data[j][i])
-                                        elif data[j][i].count(".")==1:
-                                            arr=data[j][i].split(".")
-                                            if arr[0].isnumeric() and arr[1].isnumeric():
-                                                row_dict[colnames[i]]=float(data[j][i])
-                                            else:
-                                                row_dict[colnames[i]]=data[j][i]
-                                        elif data[j][i]=='':
-                                            row_dict[colnames[i]]=None
-                                        else:
-                                            row_dict[colnames[i]]=data[j][i]
-                                    data[j]=row_dict
-                                data=data[1:]
-                                for row in data:
-                                    yield row
-        ftp.close()
+                        self.flo=flo
+    """Function to read and preprocessing data to UTF-8, converting headers to snake case and removing ~ as the quoting character in the data"""
+    def data_cleaner(self,data):
```

Comment:
Meltano SDK has this method which is usually called [`post_process`](https://sdk.meltano.com/en/latest/classes/singer_sdk.Stream.html?highlight=transform#singer_sdk.Stream.post_process)